oneview_redfish_toolkit/blueprints/storage.py

```diff
@@ -35,66 +35,84 @@
+
+    st = Storage(server_profile, server_hardware_type, sas_logical_jbods)
+
+    return ResponseBuilder.success(st)
+
+
+@storage.route("/redfish/v1/Systems/<profile_id>/Storage/1/Drives/<drive_id>",
+               methods=["GET"])
+def get_drive(profile_id, drive_id):
+    """Get the Redfish Storage for a given UUID.
+
+        Return Drive Redfish JSON for a given server profile UUID and Drive ID.
+        Logs exception of any error and return abort(500)
+        Internal Server Error.
+
+        Returns:
+            JSON: Redfish json with Storage
+            When server profile is not found calls abort(404)
+
+        Exceptions:
+            Logs the exception and call abort(500)
+
+    """
+
+    drive_id_int = None
+    logical_jbod = None
     try:
-        server_hardware = g.oneview_client.server_hardware. \
-            get(uuid)
-
-        sht_uri = server_hardware['serverHardwareTypeUri']
-        server_hardware_type = \
-            g.oneview_client.server_hardware_types.get(sht_uri)
-
-        st = Storage(uuid, server_hardware_type)
-
-        json_str = st.serialize()
-
-        return Response(
-            response=json_str,
-            status=status.HTTP_200_OK,
-            mimetype="application/json")
-    except HPOneViewException as e:
-        if e.oneview_response['errorCode'] == "RESOURCE_NOT_FOUND":
-            if e.msg.find("server-hardware-types") >= 0:
-                logging.warning(
-                    'Server hardware type ID {} not found'.
-                    format(server_hardware['serverHardwareTypeUri']))
-                abort(
-                    status.HTTP_404_NOT_FOUND,
-                    "Server hardware types not found")
-            else:
-                logging.warning(
-                    'Server hardware UUID {} not found'.
-                    format(uuid))
-                abort(
-                    status.HTTP_404_NOT_FOUND,
-                    "Server hardware not found")
-        elif e.msg.find("server-hardware-types") >= 0:
-            logging.exception(
-                'OneView Exception while looking for server hardware type'
-                ' {}'.format(e)
-            )
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-        elif e.msg.find("server-hardware") >= 0:
-            logging.exception(
-                'OneView Exception while looking for '
-                'server hardware: {}'.format(e)
-            )
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-        else:
-            logging.exception('Unexpected OneView Exception: {}'.format(e))
-            abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
-    except Exception as e:
-        # In case of error print exception and abort
-        logging.exception('Unexpected error: {}'.format(e))
-        return abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+        drive_id_int = int(drive_id)
+    except ValueError:
+        abort(status.HTTP_400_BAD_REQUEST, "Drive id should be a integer")
```

Comment:
Maybe we could log some message for this exception, for debug investigation purpose.

Reply:
We could put the log message to situations like this in generica handler to http error 400. What do you think? Because the log message will not be so different that the message of the `abort` method.

Reply:
Sure, that works for me!